ordered_model/models.py

```diff
@@ -253,15 +263,38 @@ def to(self, order, extra_update=None):
             # object is already at desired position
             return
         qs = self.get_ordering_queryset()
-        extra_update = {} if extra_update is None else extra_update
-        if getattr(self, order_field_name) > order:
-            qs.below_instance(self).above(order, inclusive=True).increase_order(
-                **extra_update
-            )
+
+        """
+        Get current value in database of our object. Necessary to prevent race conditions, where our model has been
+        moved in the database since being fetched - while it thinks it's in position 2 it's actually in position 3, 4,
+        etc. See https://github.com/bfirsh/django-ordered-model/issues/184
+        """
+        current_order_value_in_db = (
+            self.get_ordering_queryset()
+            .filter(pk=self.pk)
+            .values(self.order_field_name)
+        )
```

Comment:
this may still have other stale data in `self` that is not updated, possibly overwriting other changes. So either the final `self.save()` call should use `update_fields`, or you could call `self.refresh_from_db()` here instead of doing this query. The latter form updates _all_ the data after acquiring the lock and should bring the instance up to date again.